components/pthread/pthread.c

```diff
@@ -121,6 +123,9 @@ static esp_pthread_t *pthread_find(TaskHandle_t task_handle)
 static void pthread_delete(esp_pthread_t *pthread)
 {
     SLIST_REMOVE(&s_threads_list, pthread, esp_pthread_entry, list_node);
+    free(pthread->task_arg);
```

Comment:
this was also not freed before ? 
is this intended, or was it a bug ?

Reply:
I think you are right that this should be done together with deleting `pthread` structure. Please remove `free(pthread->task_arg);` from pthread_exit to avoid a double-free.

Reply:
done !